### PR TITLE
[PAY-728] Audio Transactions History Table

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
Audio Transactions History Table

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-728

File: packages/web/src/common/store/pages/audio-transactions/sagas.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    case TransactionType.CHALLENGE_REWARD:
+    case TransactionType.TRENDING_REWARD:
+      return {
+        signature: tx.signature,
+        transactionType: txType,
+        method: challengeMethods[tx.method],
+        date: formatDate(tx.transaction_date),
+        change: tx.change as StringAudio,
+        balance: tx.balance as StringAudio,
+        metadata: tx.metadata as unknown as string
```
Comment (piazzatron) [inline packages/web/src/common/store/pages/audio-transactions/sagas.ts:76]: making a note for later to come back and figure a better way to handle these

**Review by piazzatron — APPROVED** [review]
Looks awesome, sorry for delayed PR! Mostly just nits that we can fast follow later after you're back from vacay :)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-728

File: packages/common/src/store/pages/audio-transactions/slice.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    ) => {},
+    setAudioTransactions: (
+      state,
+      action: PayloadAction<{
+        txDetails: (TransactionDetails | {})[]
+        offset?: number
+      }>
+    ) => {
+      const { txDetails, offset } = action.payload
+      const transactionsCopy = state.transactions.slice()
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/common/src/store/pages/audio-transactions/slice.ts:50]: Using `state.transactions = state.transactions.splice(...)` breaks, don't fully understand why immer isn't able to translate that, but can use `const transactionsCopy = [...state.transactions]` instead.

File: packages/common/src/store/pages/audio-transactions/slice.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import { full } from '@audius/sdk'
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { TransactionDetails } from '../../ui/transaction-details/types'
+
+type FetchAudioTransactionsPayload = {
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/common/src/store/pages/audio-transactions/slice.ts:6]: Don't think this type is exposed from the sdk right now.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-PAY-728

File: packages/common/src/store/pages/audio-transactions/slice.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  transactionsCount: 0,
+  transactions: []
+}
+
+const slice = createSlice({
+  name: 'audio-transactions-page',
+  initialState,
+  reducers: {
+    fetchAudioTransactionsCount: () => {},
+    setAudioTransactionsCount: (
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/common/src/store/pages/audio-transactions/slice.ts:28]: Good call, but gonna punt on this for now.

File: packages/web/src/common/store/pages/audio-transactions/sagas.ts [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+function* fetchTransactionsCount() {
+  yield* takeLatest(fetchAudioTransactionsCount.type, function* () {
+    const [data, signature] = yield* call(signAuthData)
+    const response = yield* call(
+      [transactions, transactions.getAudioTransactionHistoryCount],
+      {
+        encodedDataMessage: data!,
+        encodedDataSignature: signature!
```
[the remark this answers is not in the document]
Comment (dharit-tan) [reply packages/web/src/common/store/pages/audio-transactions/sagas.ts]: Think these were possibly undefined, but changed to call `signDiscoveryNodeRequest` now.

**Pull request merged**